source/common/http/conn_manager_impl.cc

```diff
@@ -470,6 +488,60 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
       connection_manager_.config_, *snapped_route_config_, connection_manager_.random_generator_,
       connection_manager_.runtime_, connection_manager_.local_info_);
 
+  if (!cached_route_.valid()) {
+    cached_route_.value(snapped_route_config_->route(*request_headers_, stream_id_));
+  }
+
+  // Check for WebSocket upgrade request if the route is valid, and supports WebSockets.
+  if ((protocol == Protocol::Http11) && cached_route_.value()) {
```

Comment:
I was checking if the route was a null pointer. In retrospect, I can reuse the same valid() call to check if it's a nullptr.